service/common/compatibility/stateOptions.go

```diff
@@ -45,14 +45,60 @@ func GetDecideApiRetryPolicy(stateOptions *iwfidl.WorkflowStateOptions) *iwfidl.
 	return stateOptions.ExecuteApiRetryPolicy
 }
 
-func GetDataObjectsLoadingPolicy(stateOptions *iwfidl.WorkflowStateOptions) *iwfidl.PersistenceLoadingPolicy {
+func GetWaitUntilApiDataObjectsLoadingPolicy(stateOptions *iwfidl.WorkflowStateOptions) *iwfidl.PersistenceLoadingPolicy {
 	if stateOptions == nil {
 		return nil
 	}
+
+	if stateOptions.HasDataObjectsLoadingPolicy() {
+		return stateOptions.DataObjectsLoadingPolicy
+	}
+
+	if stateOptions.HasDataAttributesLoadingPolicy() {
+		return stateOptions.DataAttributesLoadingPolicy
+	}
+
+	return stateOptions.WaitUntilApiDataAttributesLoadingPolicy
+}
```

Comment:
I think the expected behavior is to prefer more specific to override less specific  -- waitUntilDALoadingPolicy will override daLoadingPolicy, then DataObjectsLoadingPolicy, then nil

Reply:
When I did this I tried copying from the existing fallback from `DataAttributesLoadingPolicy` to `HasDataObjectsLoadingPolicy`: 

```
        if stateOptions.HasDataObjectsLoadingPolicy() {
		return stateOptions.DataObjectsLoadingPolicy
	}
	return stateOptions.DataAttributesLoadingPolicy
```

But you're absolutely right, more specific should override less specific, I'll make the changes.
I will also implement a `integ` test that ensures that the more specific policy is the one being picked up if they're both specified.

Reply:
This should be done now ✅